shaders/shader.vert

```diff
@@ -0,0 +1,59 @@
+#version 310 es
+precision mediump float;
+layout(location = 0) in vec2 a_particlePos;
+layout(location = 1) in float a_particleDen;
+layout(location = 2) in vec2 a_particleVel;
+layout(location = 3) in vec2 a_pos;
+uniform mat4 u_viewProjection;
+out vec2 xsv_POSITION0;
+out vec4 COLOR0;
+
+struct Particle
+{
+    vec2 position;
+    vec2 velocity;
+};
+
+struct ParticleDensity
+{
+    float density;
+};
+
+layout(std140, binding = 0) buffer ParticlesRO
+{
+    readonly Particle particles[${g_iNumParticles}];
+} particlesRO;
+
+layout(std140, binding = 1) buffer ParticleDensityRO
+{
+    readonly ParticleDensity particles[${g_iNumParticles}];
+} particleDensityRO;
+
+vec4 Rainbow[5] = vec4[5](vec4(1, 0, 0, 1), vec4(1, 1, 0, 1), vec4(0, 1, 0, 1), vec4(0, 1, 1, 1), vec4(0, 0, 1, 1));
+
+vec4 VisualizeNumber(float n)
+{
+    return mix(Rainbow[uint(floor(n * 4.0f))], Rainbow[uint(ceil(n * 4.0f))], vec4(fract(n * 4.0f)));
+}
+
+vec4 VisualizeNumber(float n, float lower, float upper)
+{
+    return VisualizeNumber(clamp((n - lower) / (upper - lower), float(0), float(1)));
+}
+
+void main()
+{
+    xsv_POSITION0 = particlesRO.particles[gl_VertexID].position;
+    COLOR0 = VisualizeNumber(particleDensityRO.particles[gl_VertexID].density, 1000.0f, 2000.0f);
+    COLOR0 = VisualizeNumber(a_particleDen, 1000.0f, 2000.0f);
```

Comment:
It seems that this line will overwrite above line. Is there any mistake?

Reply:
Done.